api/errors/api_error.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2021 Nitric Technologies Pty Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package errors
+
+import (
+	"fmt"
+
+	"github.com/nitrictech/go-sdk/api/errors/codes"
+	"google.golang.org/grpc/status"
+)
+
+type ApiError struct {
+	code  codes.Code
+	msg   string
+	cause error
+}
+
```

Comment:
Could be useful to incorporate these Go error improvements: https://blog.golang.org/go1.13-errors
```
func (e *ApiError) Unwrap() error { return e.cause }
```

Reply:
done